### PR TITLE
#633: Error compiling file `url.replace is not a function

### PR DESCRIPTION
**Pull request by hkollmann**
- markdown error as warning
- add more tests

### CODE REVIEW

File: source/class/qx/tool/compiler/jsdoc/Parser.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ qx.Class.define("qx.tool.compiler.jsdoc.Parser", {
           try {
             cmd.body = converter.makeHtml(cmd.body);
           } catch (e) {
-            qx.tool.compiler.Console.error(`Markdown conversion error: "${e.message}" found in \n${cmd.body.trim()}`);
+            qx.tool.compiler.Console.warn(`Markdown conversion: "${e.message}" found in \n${cmd.body.trim()}`);
```
Comment (johnspackman) [inline source/class/qx/tool/compiler/jsdoc/Parser.js:102]: Isn't the exception being raised a bug in the "showdown" module?  EG what is wrong with having a comment like 

```
    /**
     * --------------------------------------------------------------------------
     * [Constructor]
     * --------------------------------------------------------------------------
     */
    construct: function()
```

Even if we decided to enforce strict rules on comment formatting, it should not say something inexplicable just `url.replace is not a function`.

Comment (cboulanger) [reply source/class/qx/tool/compiler/jsdoc/Parser.js:102]: Maybe the simplest thing would be making the warning more explicative: 
```
Markdown conversion problem: Error "${e.message}" was thrown parsing "${cmd.body.trim()}". Please review your doc comments for Markdown compatibility.
```

Comment (derrell) [reply source/class/qx/tool/compiler/jsdoc/Parser.js:102]: I don't think there should be *anything* in a comment that causes it to barf. If it's understandable as markdown, then treat it as such. If not, don't tell the user how to write a comment. (A warning that can be turned off via command-line option or `@ignore` or whatever is ok; just not an error.)

**Comment by hkollmann** [comment]
Implemented both suggestions from @cboulanger  and @derrell

**Comment by level420** [comment]
BTW: this `url.replace is not a function` seems to be a bug in node module showdown: https://github.com/showdownjs/showdown/issues/548
Created in 2018, labeled as a bug, but still unresolved :-(
So we are fighting agains a bug in another module, which is bad.

**Comment by level420** [comment]
and what I've understood from that bug is, that every markdown named like a javascript reserved word could cause the exception, like e.g. constructor

**Comment by level420** [comment]
@johnspackman as I've understood it will only break for links, but its bad anyway

**Comment by hkollmann** [comment]
Do you know an alternative to showdown?

**Comment by johnspackman** [comment]
Not personally but there are several out there.

Github use https://github.com/github/cmark-gfm, but that's a C library so there is this nodejs wrapper: https://github.com/killagu/node-cmark

For a pure JS approach there is https://github.com/markdown-it/markdown-it "Markdown parser done right. Fast and easy to extend."

Or https://marked.js.org/

**Comment by cboulanger** [comment]
@hkollmann Can you document why did you closed this? Was the bug resolved with another PR?

**Comment by hkollmann** [comment]
was an accident

**Comment by cboulanger** [comment]
@johnspackman You are right, but can we accept this workaround now and deal with switching the markdown parser later, since this would push the release date again? The current system is working except this bug. I'll create a separate issue for v7 so we don't forget. Ok?

**Comment by johnspackman** [comment]
Thank you @hkollmann

**Pull request merged**